delete.php

```diff
@@ -25,9 +25,18 @@
 require_once('../../config.php');
 
 $id = required_param('id', PARAM_INT);
+$type = optional_param('type', 'entry', PARAM_TEXT);
+
+if ($type == 'period') {
+    $dbname = 'cpdlogbook_periods';
+    $conditions = ['id' => $id];
+} else {
+    $dbname = 'cpdlogbook_entries';
+    $conditions = ['id' => $id, 'userid' => $USER->id];
+}
 
 // If the entry doesn't exist.
-$record = $DB->get_record('cpdlogbook_entries', ['id' => $id, 'userid' => $USER->id], '*', MUST_EXIST);
+$record = $DB->get_record($dbname, $conditions, '*', MUST_EXIST);
```

Comment:
Generally I would avoid abstracting away the table name, I've move this into two code paths purely make it easier to read and easier to search and grep on later.